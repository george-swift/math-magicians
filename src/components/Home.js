import { Link } from 'react-router-dom';
import Blackboard from '../assets/roman-mager.jpg';

const Home = () => (
  <div className="container flex-column text-center pt-5">
    <h1>Welcome to our page</h1>
    <div className="mt-4 card bg-dark text-white">
      <img src={Blackboard} alt="A blackboard" className="w-100" />
      <div className="card-body">
        <p className="card-text">
          Math Magicians is a single page web app that provides an easy way of solving
          arithmetic problems. Built with React, React Router v6, Javascript, Bootstrap,
          HTML5/CSS3.
        </p>
        <Link to="/calculator" className="btn btn-warning">Get Started</Link>
      </div>
    </div>
  </div>
);

export default Home;
