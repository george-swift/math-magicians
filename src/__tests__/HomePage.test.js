import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import Home from '../components/Home';

describe('Snapshot of Home Page', () => {
  test('should render page correctly', () => {
    const tree = renderer
      .create(<Router><Home /></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
