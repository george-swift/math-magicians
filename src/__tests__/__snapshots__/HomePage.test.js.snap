// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Home Page should render page correctly 1`] = `
<div
  className="container flex-column text-center pt-5"
>
  <h1>
    Welcome to our page
  </h1>
  <div
    className="mt-4 card bg-dark text-white"
  >
    <img
      alt="A blackboard"
      className="w-100"
      src="roman-mager.jpg"
    />
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        Math Magicians is a single page web app that provides an easy way of solving arithmetic problems. Built with React, React Router v6, Javascript, Bootstrap, HTML5/CSS3.
      </p>
      <a
        className="btn btn-warning"
        href="/calculator"
        onClick={[Function]}
      >
        Get Started
      </a>
    </div>
  </div>
</div>
`;
